### PR TITLE
Adding validations before redirect on user dashboard page and creating a dynamic file for rendering dashboard for all types of users

### PR DESCRIPTION
**Pull request by abhinav098**
When I was integrating most of the work, i found out that we do not need separate pages for donor, admin and recipients.
So I have created a `dashboard` page inside users directory ( a single file that will render content based on logged in user)
checkout users.js route and dashboard.handlebars file. You'll understand what i'm trying to explain.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -50,9 +50,10 @@ app.use(async (req, res, next) => {
 });
 
 // Setup loggedInUser local to use  middleware for storing user
-app.use(function (req, res, next) {
+app.use((req, res, next) => {
   if (req.session.user) {
     res.locals.loggedInUser = req.session.user;
+    res.locals.userRole = req.session.user.role;
```
Comment (abhinav098) [inline app.js:56]: You guys can use `userRole` and `loggedInUser` throughout application in handlebars without any trouble.

File: dump.js [hunk 2]
```diff
@@ -99,7 +102,7 @@ module.exports = {
       description: 'Bed Bath and Beyond area rug 37 X 60',
       region: 'Jersey City',
       zipcode: '07307',
-      donor_id: 1,
```
Comment (abhinav098) [inline dump.js:102]: fixed the dump to create donations associated to the donors.

File: middlewares/auth.js [hunk 3]
```diff
@@ -4,9 +4,10 @@ function loginRequired(req, res, next) {
 }
 
 function isLoggedIn(req, res, next) {
-  if (req.session.user) {
+  let user = req.session.user;
+  if (user) {
     req.flash('success', 'Already logged in!');
-    return res.redirect('/');
+    return res.redirect(`/users/${user._id}/dashboard`);
```
Comment (abhinav098) [inline middlewares/auth.js:10]: redirect to dashboard, if logged in.

File: routes/auth.js [hunk 4]
```diff
@@ -16,12 +16,13 @@ router.post('/login', authMiddlewares.isLoggedIn, async (req, res) => {
...
     req.session.user = user;
-    req.flash('success', 'Logged in successfully!');
     let role = await Role.findById(user.role_id);
     let role_name = role.name.toLocaleLowerCase();
+    req.session.user.role = role_name;
+    req.flash('success', 'Logged in successfully!');
 
     // redirect users to their specific dashboards
-    res.redirect(`/users/${role_name}/${user._id}`);
+    res.redirect(`/users/${user._id}/dashboard`);
```
Comment (abhinav098) [inline routes/auth.js:25]: redirect to dashboard after login

File: routes/index.js [hunk 5]
```diff
@@ -9,12 +9,16 @@ module.exports = (app) => {
...
   app.get('/', async (req, res) => {
-    res.status(200).render('static/home', {
-      title: 'Home',
-      authenticated: req.session.user ? true : false,
-      message: req.flash('success'),
-      sessionMessage: res.locals.sessionFlash,
-    });
+    let user = req.session.user;
+    if (user) {
+      return res.redirect(`/users/${user._id}/dashboard`);
```
Comment (abhinav098) [inline routes/index.js:14]: redirect to user dashboard if the user is logged in and tries to go to `/` through browser searchbar.

File: routes/users.js [hunk 6]
```diff
@@ -1,33 +1,58 @@
...
-router.get('/admin/:id', async (req, res) => {
-  let donations = await donationData.allDonations();
-  res.status(200).render('users/admin/dashboard', {
-    title: 'Admin',
-    showApproveReject: true,
-    donations,
-    message: req.flash(),
-  });
-});
+router.get('/:id/dashboard', authMiddleWare.loginRequired, async (req, res) => {
```
Comment (abhinav098) [inline routes/users.js:8]: check if the user is logged in, if not, redirect to login page.

File: routes/users.js [hunk 7]
```diff
@@ -1,33 +1,58 @@
...
+  try {
+    let sessionUser = req.session.user;
+    // search for user from id given in params
+    let searchedUser = await userData.getUserById(req.params.id);
+    if (searchedUser.role_id != sessionUser.role_id) throw 'Invalid User';
```
Comment (abhinav098) [inline routes/users.js:13]: then find and check if the correct user is set in the session and role is also correct.

File: routes/users.js [hunk 8]
```diff
@@ -1,33 +1,58 @@
...
-  let donations = await donationData.getApprovedDonations();
-  res.status(200).render('users/admin/dashboard', {
-    title: 'Recipient',
-    donations,
-    message: req.flash(),
-  });
+      options = {
+        showApproveReject: true,
+        reviewedDonations,
+        submittedDonations,
```
Comment (abhinav098) [inline routes/users.js:31]: create options and params to be passed to handlebars based on roles.

**Review by iamikenna — APPROVED** [review]

**Review by neilvarunnaidu — APPROVED** [review]

**Pull request merged**